File: app/src/main/java/com/majeur/psclient/model/Player.java
package com.majeur.psclient.model;

public enum  Player {

    FOE,
    TRAINER;

    public static Player get(String playerId, String username1, String username2, String myUsername) {
        playerId = playerId.substring(0, "pX".length());
        boolean idIsOne = playerId.contains("1");
        boolean watching = !username1.equals(myUsername) && !username2.equals(myUsername);
        if (watching)
            return idIsOne ? TRAINER : FOE;

        if (idIsOne)
            return username1.equals(myUsername) ? TRAINER : FOE;
        else
            return username2.equals(myUsername) ? TRAINER : FOE;
    }

    public String username(String username1, String username2, String myUsername) {
        if (username1.equals(myUsername)) {
            return this == TRAINER ? username1 : username2;
        } else if (username2.equals(myUsername)) {
            return this == TRAINER ? username2 : username1;
        } else {
            return this == TRAINER ? username1 : username2;
        }
    }

}

File: psclient/src/main/java/com/majeur/psclient/model/battle/Player.kt
package com.majeur.psclient.model.battle


enum class Player {

    FOE, TRAINER;

    fun username(username1: String, username2: String, myUsername: String): String {
        return if (username1 == myUsername) {
            if (this == TRAINER) username1 else username2
        } else if (username2 == myUsername) {
            if (this == TRAINER) username2 else username1
        } else {
            if (this == TRAINER) username1 else username2
        }
    }

    companion object {
        operator fun get(playerId: String, username1: String?, username2: String?, myUsername: String): Player {
            val isOne = playerId.take(2).contains("p1", ignoreCase = true)
            val watching = username1 != myUsername && username2 != myUsername
            if (watching) return if (isOne) TRAINER else FOE
            return if (isOne) if (username1 == myUsername) TRAINER else FOE else if (username2 == myUsername) TRAINER else FOE
        }
    }
}
